feat: support streameable http transport

alejandropta opened the pull request:
### Description

Includes the new Streamable HTTP transport

### Test Procedure

Created a Remote MCP with new protocol and tested it

### Type of Change

<!-- Put an 'x' in all boxes that apply -->

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [x] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

<!-- Put an 'x' in all boxes that apply -->

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [x] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [x] I have created a changeset using `npm run changeset` (required for user-facing changes)
-   [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Screenshots

<img width="387" alt="image" src="https://github.com/user-attachments/assets/5b41e980-bcc0-42f5-bb3b-ee023d8c7e59" />

### Additional Notes

I need to use discriminatedUnion because streamable-http and sse colided in shape of their zod definitions

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Adds support for streamable HTTP transport in MCP framework with updated configurations and dependencies.
> 
>   - **Behavior**:
>     - Adds `StreamableHTTPClientTransport` to `McpHub.ts` for handling streamable HTTP transport.
>     - Updates `McpConnection` and `McpTransportType` to include `streamable-http`.
>     - Modifies `connectToServer()` in `McpHub.ts` to handle `streamable-http` transport type.
>   - **Configuration**:
>     - Adds `StreamableHTTPConfigSchema` to `McpHub.ts` for streamable HTTP configuration.
>     - Updates `ServerConfigSchema` to use `z.discriminatedUnion` for transport types.
>   - **Dependencies**:
>     - Updates `@modelcontextprotocol/sdk` version in `package.json` to `^1.11.1`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 509f40dd4b029426a2056dd57ed3041e7de51d2c. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

RittikaJindalTR commented:
Can we get this PR merged?

pashpashpash reviewed (dismissed):
Thanks @alejandropta for this PR!

This looks great - adding support for streamable HTTP transport is a nice addition to our MCP framework. I like how you've:

1. Followed the existing patterns in the codebase
2. Used discriminatedUnion for better type safety between transport schemas
3. Properly updated the connectToServer method to handle the new transport type
4. Included a proper changeset for the minor version bump

The implementation is clean and doesn't break existing functionality. The SDK version update to 1.11.1 makes sense for this feature.

Great work!

arafatkatze commented on esbuild.js:
```diff
@@ -134,14 +134,6 @@ const baseConfig = {
 		aliasResolverPlugin,
 		/* add to the end of plugins array */
 		esbuildProblemMatcherPlugin,
-		{
```
Can you explain this?

alejandropta commented:
Thanks for the review @pashpashpash I have updated the branch with main as well and I've seen now I need to support the method connectToServerRPC, last commit includes this change so it is available as well. I've have also changed from the literal "streameable-http" to simple "http" as it is easier to type and more ux friendly for everyone
